geopy/geocoders/banfrance.py

```diff
@@ -0,0 +1,194 @@
+
+        :param str domain: Currently it is ``'api-adresse.data.gouv.fr'``, can
+            be changed for testing purposes.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+            .. versionadded:: 1.12.0
+
+        :param str format_string:
+            See :attr:`geopy.geocoders.options.default_format_string`.
+
+            .. versionadded:: 1.14.0
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+            .. versionadded:: 1.14.0
+
+        """
+        super(BANFrance, self).__init__(
+            format_string=format_string,
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+        )
+        self.domain = domain.strip('/')
+
+        self.geocode_api = (
+            '%s://%s%s' % (self.scheme, self.domain, self.geocode_path)
+        )
+        self.reverse_api = (
+            '%s://%s%s' % (self.scheme, self.domain, self.reverse_path)
+        )
+
+    def geocode(
+            self,
+            query,
+            limit=None,
+            exactly_one=True,
+            timeout=DEFAULT_SENTINEL,
+    ):
+        """
+        Return a location point by address.
+
+        :param str query: The address or query you wish to geocode.
+
+        :param int limit: Defines the maximum number of items in the
+            response structure. If not provided and there are multiple
+            results the BAN API will return 5 results by default.
+            This will be reset to one if ``exactly_one`` is True.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+
+        """
+        params = {
+            'q': self.format_string % query,
+        }
+
+        url = "?".join((self.geocode_api, urlencode(params)))
+
+        logger.debug("%s.geocode: %s", self.__class__.__name__, url)
+        return self._parse_json(
+            self._call_geocoder(url, timeout=timeout), exactly_one
+        )
+
+    def reverse(
+            self,
+            query,
+            exactly_one=DEFAULT_SENTINEL,
```

Comment:
It should be just `exactly_one=True`, and the corresponding deprecation warning is unnecessary.